### PR TITLE
Refactor Sprint 13

### PR DESCRIPTION
**Pull request by Marvin-Beckmann**
**Description**

## **!Still To Be Discussed!**

- [x] consistent variable naming
- [x] new notation for tuples -> (a, b): (A,B)
- [x] Update interfaces using impl Into<Modulus>
- [x] Whitespaces after ","
- [x] Allow whitespace in matrices -> "], ["
- [x] big -> large
- [x] consistently end assert lines with ; for tests

**Testing**

No new functionality has been added and all previous tests still work

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/rational/q/from.rs [hunk 1]
```diff
@@ -193,11 +193,11 @@ impl<IntegerNumerator: AsInteger, IntegerDenominator: AsInteger>
...
     /// Create a [`Q`] from two integers.
     /// The integer types can be, for example, [`Z`],
-    /// [`Zq`](crate::integer_mod_q), [`u32`], [`i64`] or references to these types
+    /// [`Zq`](crate::integer_mod_q), [`u32`],[`i64`] or references to these types
     ///
     /// Parameters:
-    /// - `num_den_tuple` is a tuple of integers `(numerator, denominator)`
-    ///   The first and second element of the tuple may have different integer types.
+    /// - `num`: The value of the numerator
+    /// - `den`: The value of the denominator
```
Comment (Marcel583) [inline src/rational/q/from.rs:200]:
```suggestion
    /// - `num`: The value of the numerator.
    /// - `den`: The value of the denominator.
```

File: src/utils/parse.rs [hunk 2]
```diff
@@ -51,6 +51,7 @@ pub(crate) fn parse_matrix_string(string: &str) -> Result<Vec<Vec<String>>, Math
     }
 
     // delete `[[` in front and `]]` in the end and split the matrix into rows
+    let string = string.replace("], [", "],[");
```
Comment (Marcel583) [inline src/utils/parse.rs:54]: \s* in the RegEx  means that arbitrarily many spaces may occur, but this still restricts us to one space and may lead to errors as matrices go through the matching that are not valid.

File: src/traits.rs [hunk 3]
```diff
@@ -232,7 +232,7 @@ pub(crate) unsafe trait AsInteger {
     ///
     /// It is intended to be used when a read only [`fmpz`] reference is required
     /// for a Flint function call.
-    /// If the data type does not contain a [`fmpz`], [`into_fmpz`](AsInteger::into_fmpz)
+    /// If the data type does not contain a [`fmpz`],[`into_fmpz`](AsInteger::into_fmpz)
```
Comment (Marcel583) [inline src/traits.rs:235]: I would leave this space here

File: src/rational/q/arithmetic/logarithm.rs [hunk 4]
```diff
@@ -81,9 +81,9 @@ impl Q {
 
         let ln_value = self.ln()?;
         let ln_base = base.ln()?;
-        let log_b = ln_value / ln_base;
+        let log_2 = ln_value / ln_base;
 
-        Ok(log_b)
+        Ok(log_2)
```
Comment (Marcel583) [inline src/rational/q/arithmetic/logarithm.rs:86]: As above

**Review by Marcel583 — CHANGES_REQUESTED** [review]

File: src/integer_mod_q/z_q/from.rs [hunk 5]
```diff
@@ -34,7 +34,7 @@ impl Zq {
     /// let value = Z::from(42);
     /// let modulus = Modulus::from(100);
     ///
-    /// let answer_a = Zq::from_z_modulus(&value, &modulus);
+    /// let answer_1 = Zq::from_z_modulus(&value, &modulus);
```
[the remark this answers is not in the document]
Comment (Marvin-Beckmann) [reply src/integer_mod_q/z_q/from.rs:37]: adapted it to value_zq -> seemed to fit better

File: src/utils/parse.rs (hunk 2 above)
Comment (Marvin-Beckmann) [reply src/utils/parse.rs:54]: replaced it with \s?

**Review by Marcel583 — APPROVED** [review]

**Pull request merged**